roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+import h5py
+from lazy_ops import DatasetView
+from ...segmentationextractor import SegmentationExtractor
+from ...extraction_tools import _pixel_mask_extractor
+import os
+
+class CaimanSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'CNMF-E\' ROI segmentation method.
+    """
+    extractor_name = 'CaimanSegmentation'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._roi_response_fluorescence = self._roi_response
+        self._roi_response_neuropil = self._trace_extractor_read('C')
+        self._roi_response_deconvolved = self._trace_extractor_read('S')
+        self._images_correlation = self._summary_image_read()
+        self._raw_movie_file_location = self._dataset_file['params']['data']['fnames'][0].decode('utf-8')
+        self._sampling_frequency = self._dataset_file['params']['data']['fr'][()]
+        self.image_masks = None
```

Comment:
can we set this in the init?

Reply:
In the caiman file, the `image_masks `are stored as a sparse matrix. Since there is no way to lazily reshape and create an array out of the sparse matrix would load the whole thing into memory. 
I'm trying to avoid this so I set image_masks as None. Would it then be better to have image_masks as an internal attribute for all classes then? 

I'm calling `self._image_mask_sparse_read()` within the `get_image_masks()` which only returns `image_masks` for the requested ROI ids, thus possibly loading a smaller value into memory.

Reply:
let's just not make it lazy for now

Reply:
hmm, okay, fixed that https://github.com/catalystneuro/roiextractors/pull/34/commits/5e3fd1678ae152fb198b8958dc3f5d593e94dacd